tests/s3/copy_object/test_copy_object_quota.py

```diff
@@ -0,0 +1,724 @@
+        self.log.info("Step 1: Create a user (user1)")
+        payload = self.csm_obj.iam_user_payload_rgw("random")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        self.user_id = resp.json()["keys"][0]['user']
+        self.created_iam_users.add(self.user_id)
+        resp1 = self.csm_obj.compare_iam_payload_response(resp, payload)
+        self.log.info("Printing response %s", resp1)
+        assert resp1[0], resp1[1]
+        self.akey = resp.json()["keys"][0]["access_key"]
+        self.skey = resp.json()["keys"][0]["secret_key"]
+        self.src_bkt = "src1"
+        self.obj = "obj"
+        self.dest_bkt1 = "dest1"
+        self.log.info("Step 2: Create bucket under above IAM user")
+        self.log.info("Verify Create bucket: %s with access key: %s and secret key: %s",
+                      self.src_bkt, self.akey, self.skey)
+        bucket_created = s3_misc.create_bucket(self.src_bkt, self.akey, self.skey)
+        assert bucket_created, "Failed to create bucket"
+        self.buckets_created.append([self.src_bkt, self.akey, self.skey])
+        self.log.info("ENDED: Test setup.")
+
+    def teardown_method(self):
+        """
+        Teardown will be invoked after running each test case.
+
+        It will clean all resources which are getting created during
+        test execution such as S3 buckets and the objects present into that bucket.
+        """
+        self.log.info("STARTED: Test teardown.")
+        self.log.info("Deleting buckets %s & associated objects", self.buckets_created)
+        buckets_deleted = []
+        iam_deleted = []
+        for bucket in self.buckets_created:
+            resp = s3_misc.delete_objects_bucket(bucket[0], bucket[1], bucket[2])
+            if resp:
+                buckets_deleted.append(bucket)
+            else:
+                self.log.error("Bucket deletion failed for %s ", bucket)
+        self.log.info("buckets deleted %s", buckets_deleted)
+        for bucket in buckets_deleted:
+            self.buckets_created.remove(bucket)
+
+        self.log.info("Deleting iam account %s created in test", self.created_iam_users)
+        for iam_user in self.created_iam_users:
+            resp = self.csm_obj.delete_iam_user(iam_user)
+            if resp.status_code == HTTPStatus.OK:
+                iam_deleted.append(iam_user)
+            else:
+                self.log.error("IAM deletion failed for %s ", iam_user)
+        self.log.info("IAMs deleted %s", iam_deleted)
+        for iam in iam_deleted:
+            self.created_iam_users.remove(iam)
+        assert_utils.assert_true(len(self.buckets_created) == 0, "Bucket deletion failed")
+        assert_utils.assert_true(len(self.created_iam_users) == 0, "IAM deletion failed")
+        self.log.info("ENDED: Test teardown.")
+
+    def create_iam_user(self):
```

Comment:
This can be moved to s3_misc

Reply:
S3misc is more related to S3APIs creation with specific access_key and secret_key and other parameter instead of using s3Lib functions with default accesskey and secrete key.
On the other hand, create_iam_user is a wrapper, integrating steps required for iam user creation.

Reply:
I found an existing legacy create_iam_user in s3_misc, so it would be better to consolidate your code there rather than have duplication of helper/lib methods. You can check the S3_ENGINE and execute the appropriate code block, so that legacy LC code is not lost

Reply:
legacy s3-iam user is differnt than rgw s3-iam user. Please check both code snippet

Reply:
Yes, which is why I had mentioned you can check S3_ENGINE within the function and retain legacy code, as well. You can move your code to the code block after checking S3_ENGINE is 1 or the constant S3_ENGINE_CORTX